app/main.py

```diff
@@ -1,2 +1,89 @@
+from __future__ import annotations
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.capacity = 8
+        self.loadfactor = 2 / 3
+        self.size = 0
+        self.hash_table = [None] * self.capacity
+        self.keys = []
+
+    def add_hash_node(self, key: Any, value: Any) -> None:
+        hash_num = hash(key)
+        hash_slot = hash_num % self.capacity
+        if self.hash_table[hash_slot] is None:
+            self.hash_table[hash_slot] = (key, hash_num, value)
+        else:
+            while self.hash_table[hash_slot] is not None:
+                if hash_slot == self.capacity - 1:
+                    hash_slot = 0
+                else:
+                    hash_slot += 1
+            self.hash_table[hash_slot] = (key, hash_num, value)
+
+    def resize_hash_table(self) -> None:
+        self.hash_table += [None] * self.capacity
+        self.capacity = self.capacity * 2
+        for index, hash_node in enumerate(self.hash_table):
+            if hash_node is not None:
+                key, hash_num, value = hash_node
+                self.hash_table[index] = None
+                self.add_hash_node(key, value)
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        hash_num = hash(key)
+        try:
+            self.__getitem__(key)
+        except KeyError:
+            self.size += 1
+            if self.size <= self.capacity * self.loadfactor:
+                self.add_hash_node(key, value)
+            else:
+                self.resize_hash_table()
+                self.add_hash_node(key, value)
+            self.keys.append(key)
+        old_value = self.__getitem__(key)
+        node_index = self.hash_table.index((key, hash_num, old_value))
+        self.hash_table[node_index] = (key, hash_num, value)
+
+    def __getitem__(self, key: Any) -> Any:
+        if key in self.keys:
+            hash_slot = hash(key) % self.capacity
+            node = self.hash_table[hash_slot]
+            if node is not None and node[0] == key:
+                return node[2]
+            for index, node in enumerate(self.hash_table):
+                if node is not None and node[0] == key:
+                    return node[2]
+        raise KeyError("Key is not in a dictionary")
+
+    def __len__(self) -> int:
+        return self.size
+
+    def __iter__(self) -> Dictionary:
+        self.index = 0
+        return self
+
+    def __next__(self) -> Any:
+        index = self.index
+        if index >= len(self.keys):
+            raise StopIteration
+        self.index += 1
+        return self.keys[index]
+
+    def get(self, key: Any, value: Any = None) -> Any:
+        try:
+            self.__getitem__(key)
+        except KeyError:
+            if value is None:
+                return None
+            self.__setitem__(key, value)
```

Comment:
It is not logical. See screenshot
<img width="277" alt="image" src="https://user-images.githubusercontent.com/65166203/214571375-3287ebeb-fa11-4a74-994a-254b2a312676.png">

Reply:
result will be:
b
b
But must return
b
None